Eliminate test flakiness

https://github.com/soypat/lneto/actions/runs/30654356364/job/91235064664
```
Run go test -race -shuffle=on -count=1 -timeout=10m ./...
ok  	github.com/soypat/lneto	1.032s
ok  	github.com/soypat/lneto/arp	1.010s
ok  	github.com/soypat/lneto/dhcp/dhcpv4	1.011s
ok  	github.com/soypat/lneto/dhcp/dhcpv6	1.009s
ok  	github.com/soypat/lneto/dns	1.008s
ok  	github.com/soypat/lneto/dns/mdns	1.008s
ok  	github.com/soypat/lneto/ethernet	1.011s
?   	github.com/soypat/lneto/examples/berkeley-listener	[no test files]
?   	github.com/soypat/lneto/examples/gen/gen-binary-bench	[no test files]
?   	github.com/soypat/lneto/examples/httphi-fuzz	[no test files]
?   	github.com/soypat/lneto/examples/httphi-over-ethernet	[no test files]
?   	github.com/soypat/lneto/examples/httphi-server	[no test files]
?   	github.com/soypat/lneto/examples/httpraw-client	[no test files]
?   	github.com/soypat/lneto/examples/httptap	[no test files]
?   	github.com/soypat/lneto/examples/min-working-example	[no test files]
?   	github.com/soypat/lneto/examples/ntp-client	[no test files]
?   	github.com/soypat/lneto/examples/ntp-server	[no test files]
?   	github.com/soypat/lneto/examples/xcurl	[no test files]
ok  	github.com/soypat/lneto/http/httphi	1.055s
ok  	github.com/soypat/lneto/http/httpraw	1.029s
ok  	github.com/soypat/lneto/internal	1.826s
?   	github.com/soypat/lneto/internal/benchci	[no test files]
?   	github.com/soypat/lneto/internal/ltesto	[no test files]
ok  	github.com/soypat/lneto/internet	1.014s
ok  	github.com/soypat/lneto/internet/pcap	1.010s
ok  	github.com/soypat/lneto/ipv4	1.009s
ok  	github.com/soypat/lneto/ipv4/icmpv4	1.008s
ok  	github.com/soypat/lneto/ipv4/linklocal4	1.009s
ok  	github.com/soypat/lneto/ipv6	1.009s
ok  	github.com/soypat/lneto/ipv6/icmpv6	1.008s
ok  	github.com/soypat/lneto/ntp	1.013s
?   	github.com/soypat/lneto/phy	[no test files]
ok  	github.com/soypat/lneto/tcp	1.994s
ok  	github.com/soypat/lneto/udp	1.011s
-test.shuffle 1785521724498944000
--- FAIL: TestRunAsyncStress (0.02s)
    runner_test.go:521: reordered: seq 1 after 3
FAIL
FAIL	github.com/soypat/lneto/x/netdev	0.184s
ok  	github.com/soypat/lneto/x/nts	1.030s
?   	github.com/soypat/lneto/x/rawsock	[no test files]
ok  	github.com/soypat/lneto/x/xnet	4.532s
FAIL
Error: Process completed with exit code 1.
```

--- a/x/netdev/buffer.go
+++ b/x/netdev/buffer.go
@@ -107,9 +107,30 @@ func (bs *bufferSelect) numFree() (numFree int) {
 }
 
 // getRx returns the oldest published Rx frame, or nil if none is pending.
+//
+// The slot scan is not a consistent snapshot: goroPutRx may publish a frame
+// into an already-scanned slot while this scan is in progress. Because the
+// producer publishes frames in seq (arrival) order, any such straggler carries
+// a lower seq than the candidate and must be delivered first to preserve
+// arrival order. A confirming re-scan detects it; the loop retries until no
+// older frame is observed, which terminates because the candidate seq strictly
+// decreases and is bounded below by the true oldest pending frame.
 func (bs *bufferSelect) getRx() []byte {
-	oldest := -1
-	var oldestSeq uint32
+	for {
+		oldest, oldestSeq := bs.scanOldest()
+		if oldest < 0 {
+			return nil
+		}
+		if !bs.hasPendingOlderThan(oldestSeq) {
+			return bs.bufs[oldest].buf[:bs.bufs[oldest].lenAcquire.Load()]
+		}
+	}
+}
+
+// scanOldest returns the index and seq of the pending Rx frame with the lowest
+// arrival seq, or -1 if none is pending.
+func (bs *bufferSelect) scanOldest() (oldest int, oldestSeq uint32) {
+	oldest = -1
 	for i := range bs.bufs {
 		n := bs.bufs[i].lenAcquire.Load()
 		if n > 0 && bs.bufs[i].isRx.Load() &&
@@ -118,10 +139,19 @@ func (bs *bufferSelect) getRx() []byte {
 			oldestSeq = bs.bufs[i].seq
 		}
 	}
-	if oldest < 0 {
-		return nil
+	return oldest, oldestSeq
+}
+
+// hasPendingOlderThan reports whether any pending Rx frame has a seq strictly
+// less than seq, i.e. a frame that should be delivered before it.
+func (bs *bufferSelect) hasPendingOlderThan(seq uint32) bool {
+	for i := range bs.bufs {
+		n := bs.bufs[i].lenAcquire.Load()
+		if n > 0 && bs.bufs[i].isRx.Load() && lessThan(bs.bufs[i].seq, seq) {
+			return true
+		}
 	}
-	return bs.bufs[oldest].buf[:bs.bufs[oldest].lenAcquire.Load()]
+	return false
 }
 
 func (bs *bufferSelect) release(buf []byte) {

